data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/OpenSearchSink.java

```diff
@@ -498,31 +495,17 @@ SerializedJson getDocument(final Event event) {
     }
 
     String routingValue = null;
-    if (routingField != null) {
-        routingValue = event.get(routingField, String.class);
-    } else if (routing != null) {
+    if (routing != null) {
       try {
         routingValue = event.formatString(routing, expressionEvaluator);
       } catch (final ExpressionEvaluationException | EventKeyNotFoundException e) {
         LOG.error("Unable to construct routing with format {}, the routing will be generated by OpenSearch", routing, e);
       }
     }
 
-    String pipelineValue = null;
-    if (pipeline != null) {
```

Comment:
Why is this being removed? You still need this because "pipelineName" can be an expression.

Reply:
In the original implementation, IndexConfiguration.getPipeline() would always return null since it pulled form a non existent field in pipelineSettings, meaning the pipelineValue for objects in bulkRequest was always null. If pipelineValue is populated, the bulkRequest to open Search would throw error and fail.  So bulkRequest only works if pipelineValue is set to null, and in the original implementation it was already null.